src/test/ui/impl-trait/bound-normalization.rs

```diff
@@ -0,0 +1,102 @@
+// compile-fail
```

Comment:
Please split this test up into several files (at least in terms of OK/ERROR) and move at least the `async-await` part to `src/test/ui/async-await`.